packages/relay/src/lib/eth.ts

```diff
@@ -1583,7 +1583,7 @@ export class EthImpl implements Eth {
       return null;
     }
   
-    if (blockNumberOrTag != null && blockNumberOrTag.length < 32 && !EthImpl.blockTagIsEarliestOrPending(blockNumberOrTag)) {
+    if (blockNumberOrTag != null && blockNumberOrTag.length < 32 && !EthImpl.blockTagIsEarliestOrPending(blockNumberOrTag) && !EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
```

Comment:
Should be a test to confirm behaviour here.
The fact that there's no failing tests means there's missing test coverage

Reply:
Done.